### PR TITLE
Piggy mob development

### PR DESCRIPTION
**Pull request by elizaMathaai**
User interests  with Checkboxes and PATCH

### CODE REVIEW

File: src/components/Interests.js [hunk 1]
```diff
@@ -1,32 +1,117 @@
...
+        console.log(interests);
+        let url = "http://192.168.43.102:8083/user/interest/"
+        await fetch(`${url}${this.state.userid}`,{
+                         method: 'PATCH',
+                         headers: {
+                            'Accept': 'application/json',
+                            'Content-Type': 'application/json',
+                            },
+                         body: JSON.stringify({
+                           "user_interests" : interests
```
Comment (rajan123456) [inline src/components/Interests.js]: There is a change pushed in user service for user_interests. It now expects a ArrayList<String>. Please test again and change accordingly.

Comment (elizaMathaai) [reply src/components/Interests.js]: Updated. passes interests as array.

File: src/components/Fblogin.js [hunk 2]
```diff
@@ -9,97 +9,136 @@ export default class Fblogin extends Component {
...
+       </View>
     );
   }
 
-
-};
+  sendUpdate = async () => {
+          const {fbData} = this.state;
+          try{
+              let response = await fetch('http://192.168.43.102:8083/user/create',{
```
Comment (rajan123456) [inline src/components/Fblogin.js]: Use the GCP IP alone. Define in a config file and use everywhere.

**Comment by piggy1-mvn** [comment]
[Mobile App: User Preferences](https://trello.com/c/qFMkXkHd/31-mobile-app-user-preferences)

**Comment by piggy1-mvn** [comment]
[Mobile App navigation depending on user logged in or not.](https://trello.com/c/zlWajBPo/59-mobile-app-navigation-depending-on-user-logged-in-or-not)

**Comment by piggy1-mvn** [comment]
[Mobile App: Integrate with Firebase to receive push notifications](https://trello.com/c/etsX9eeT/65-mobile-app-integrate-with-firebase-to-receive-push-notifications)

**Comment by piggy1-mvn** [comment]
[Mobile App: JWT Token for FB users](https://trello.com/c/sZj6K2rO/96-mobile-app-jwt-token-for-fb-users)

File: src/components/Interests.js [hunk 3]
```diff
@@ -1,32 +1,131 @@
...
+        this.setState({ options: options });
+
+      }
+
+
+     updateInterests = async () => {
+
+        let interests = this.state.options.filter(item =>item.checked).map((item=>item.title));
+        //let url = "http://192.168.43.102:8083/user/interest/"
+        let url = "http://35.222.231.249:8083/user/interest/"
```
Comment (piggy1-mvn) [inline src/components/Interests.js]: Use base url here

File: src/components/LoginPage.js [hunk 4]
```diff
@@ -19,43 +17,37 @@ export default class LoginPage extends Component{
...
-       //if(userInfo.email_id === this.state.email_id && userInfo.password === this.state.password){
-         // alert('You have successfully logged in  to PiggyBack !!');
-          //await AsyncStorage.setItem('isLoggedIn', '1');
-          //this.props.navigation.navigate('Home');
-          //} else {
-           //alert('Email id or password entered is wrong !!');
-          //}
+       try{
+       //http://192.168.43.102:8083/user/login
+           let response = await fetch('http://35.222.231.249:8083/user/login',{
```
Comment (piggy1-mvn) [inline src/components/LoginPage.js]: Use baseUrl here

File: src/components/RegisterPage.js [hunk 5]
```diff
@@ -20,12 +20,14 @@ export default class RegisterPage extends Component{
...
+
     }
 
+
     onRegisterPressed = async () => {
-      console.log("inside onregisterpressedd");
       try{
-        let response = await fetch('http://192.168.43.102:8083/user',{
+      //http://192.168.43.102:8083/user/create
+        let response = await fetch('http://35.222.231.249:8083/user/create',{
```
Comment (piggy1-mvn) [inline src/components/RegisterPage.js]: Use baseUrl here

File: src/config/Config.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export const baseUrlUserApi = "http://192.168.43.102:8083/user/";
+export const baseUrlPartnerApi = "http://192.168.43.102:8085/partner";
+export const baseUrlLocationApi = "http://192.168.43.102:8080/location";
```
Comment (piggy1-mvn) [inline src/config/Config.js]: Update IP addresses to GCP IP

File: src/config/Config.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+export const toastDuration = 3000;
+export const extendUserSessionMinutes = 10;
```
Comment (piggy1-mvn) [inline src/config/Config.js]: Do not include unused constants in config

File: src/services/LocationTracker.js [hunk 8]
```diff
@@ -92,15 +121,15 @@ export default class LocationTracker extends Component<{}> {
   }
 
   handlePress = async () => {
-    console.log("in fetch call")
+
     fetch('http://192.168.43.102:8080/location',{
```
Comment (piggy1-mvn) [inline src/services/LocationTracker.js]: Use baseUrl here

**Pull request merged**